WordPress/Classes/Models/PostCategory+Creation.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+
+extension PostCategory {
+
+    static func create(withBlogID id: NSManagedObjectID, in context: NSManagedObjectContext) throws -> PostCategory {
+        let object = try context.existingObject(with: id)
+
+        guard let blog = object as? Blog else {
+            fatalError("The object id does not belong to a Blog: \(id)")
```

Comment:
Did you consider throwing an error here, instead of crashing?

However, I'm guessing the rationale for crashing is that it's impossible at runtime not to receive a `Blog` from the `NSManagedObjectContext` call, and that the typcast is necessary only because of the Objective-C legacy in Core Data?

Reply:
I think it's still possible that the app goes into this `fatalError` branch, because there is nothing preventing us from passing any random `NSManagedObjectID` to this function, which means the `object` can be any random type, which is a programming error IMO.

Reply:
> there is nothing preventing us from passing any random NSManagedObjectID to this function, which means the object can be any random type, which is a programming error IMO.

Yeah. Fair enough, I guess "impossible" was not the right word to use.